app/views/email_authentications/passwords/edit.html.erb

```diff
@@ -21,17 +22,20 @@
           <li id="needs_lower"><i class="fas fa-check-circle"></i><%= t("devise.password_creation.needs_lower_case") %></li>
           <li id="needs_upper"><i class="fas fa-check-circle"></i><%= t("devise.password_creation.needs_upper_case") %></li>
         </ul>
-      <%= f.primary t("devise.password_creation.reset_button"), id: "password-submit", disabled: true %>
-    <% end %>
+        <%= f.primary t("devise.password_creation.reset_button"), id: "password-submit", disabled: true %>
+      <% end %>
+    </div>
   </div>
-</div>
 
-<div class="actions mt-4">
-  <%= render "email_authentications/shared/links" %>
-</div>
+  <div class="actions mt-4">
+    <%= render "email_authentications/shared/links" %>
+  </div>
 
-<script>
-  new PasswordVisibilityToggle();
-  const validator = new PasswordValidation();
-  validator.initialize();
-</script>
\ No newline at end of file
+  <script>
+    new PasswordVisibilityToggle();
+    const validator = new PasswordValidation();
+    validator.initialize();
+  </script>
+<% else %>
+  <%= render partial: "expired_reset_token" %>
+<% end %>
```

Comment:
Rather than a big conditional in the view, I think overriding the controller is better - and fairly straightforward too.

We can introduce an `EmailAuthentications::PasswordsController` with:
```ruby
def edit
  if resource.reset_password_period_valid?
    super
  else
    render :expired_reset_token
  end
end
```

And we can convert the `expired_reset_token` partial into a view.

Reply:
Music to my ears. I thought I would get pushback on subclassing the controller, but now that I have some buy-in... You may have noticed my comment in the "testing" instructions. The current flow is not ideal because you can't know if the link is invalid until the use posts. This should (hopefully) allow me to fix that. I'll give it a try. Thanks!